bids-validator/utils/issues/list.js

```diff
@@ -1057,31 +1057,63 @@ export default {
   216: {
     key: 'SAMPLE_ID_COLUMN',
     severity: 'error',
-    reason:
-      "Samples .tsv files must have a 'sample_id' column.",
+    reason: "Samples .tsv files must have a 'sample_id' column.",
   },
   217: {
     key: 'PARTICIPANT_ID_COLUMN',
     severity: 'error',
-    reason:
-      "Samples .tsv files must have a 'participant_id' column.",
+    reason: "Samples .tsv files must have a 'participant_id' column.",
   },
   218: {
     key: 'SAMPLE_TYPE_COLUMN',
     severity: 'error',
-    reason:
-      "Samples .tsv files must have a 'sample_type' column.",
+    reason: "Samples .tsv files must have a 'sample_type' column.",
   },
   219: {
     key: 'SAMPLE_TYPE_VALUE',
     severity: 'error',
     reason:
-      "sample_type MUST consist of one of the following values: cell line, in vitro differentiated cells, primary cell, cell-free sample, cloning host, tissue, whole organisms, organoid or technical sample.",
+      'sample_type MUST consist of one of the following values: cell line, in vitro differentiated cells, primary cell, cell-free sample, cloning host, tissue, whole organisms, organoid or technical sample.',
   },
   220: {
     key: 'SAMPLE_ID_DUPLICATE',
     severity: 'error',
     reason:
-      "Each sample from a same subject MUST be described by one and only one row.",
-  }
+      'Each sample from a same subject MUST be described by one and only one row.',
+  },
+  221: {
+    key: 'PIXEL_SIZE_INCONSISTENT',
+    severity: 'error',
+    reason:
+      'PixelSize need to be consistent with PhysicalSizeX, PhysicalSizeY and PhysicalSizeZ OME metadata fields',
+  },
+  222: {
+    key: 'INVALID_PIXEL_SIZE_UNIT',
+    severity: 'error',
+    reason: 'PixelSize consistency is only validated for "mm", "µm" and "nm".',
+  },
+  223: {
+    key: 'NO_VALID_JSON_FILES',
+    severity: 'error',
+    reason:
+      'No enough JSON files used to check PixelSize OME metadata fields consistency',
```

Comment:
Based on my test, if an ometiff file doesn't have a corresponding json file, there's no error raised. I think `PixelSize` and `pixelSizeUnits` are only validated if there's a JSON file to be used to check.

Reply:
Interesting! Thanks for checking.
In that case, we should first have checks to confirm that `PixelSize` and `PixelSizeUnits` are present, for all files (PNG, TIF and OME-TIFF). Then we will not need this specific check afterwards.

I found an example with RepetitionTime [here](https://github.com/neuropoly/bids-validator/blob/59a0bf89e6e70a7a680448e37ea8affc0fcdf4ef/bids-validator/utils/issues/list.js#L75).
It's in MRI and would correspond to [this](https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#required-fields) in the spec.

Reply:
I think if there's no corresponding json file, both `PixelSize` and `PixelSizeUnits` will not be checked even they are marked as required. On the other hand, the json file will be used to check if they are present. In this case, if we check `PixelSize` and `PixelSizeUnits` in our code again, I think it's unnecessary.  Instead of checking these two fields, we should check if there's no corresponding json file.

Reply:
Ok, I get it I think!
I agree that we should check that each files have a corresponding json.
That must be true for all OME-TIFF, PNG and TIF files (microscopy only), and for all suffixes except` _photo`.

Reply:
I would also suggest the following message:
'No valid JSON file found for this file.'

Reply:
Done